src/coreclr/nativeaot/Runtime/portable.cpp

```diff
@@ -368,6 +368,18 @@ COOP_PINVOKE_HELPER(Object *, RhpCheckedXchg, (Object ** location, Object * valu
     return ret;
 }
 
+COOP_PINVOKE_HELPER(uint8_t, RhpLockCmpXchg8, (uint8_t * location, uint8_t value, uint8_t comparand))
+{
+    ASSERT_UNCONDITIONALLY("NYI");
```

Comment:
This will create work for wasm in runtimelab once these APIs get used. Can this be implemented in this PR? (It should be just copy & paste in a few places.)

Reply:
I've partially reintroduced the helper methods after @SingleAccretion asked me to do so for WASM (and RISC-V) but I've left them NYI on purpose since they require the raw assembly helpers I've deleted here on XArch and ARM64 paths (after making them always expand) for debugging support and implementing them here without adding those would create an untested code path which would still require changes on the LLVM port for those WASM helpers.

Reply:
Ok, these can be implemented in separate PR.

Reply:
FWIW, the downstream perspective here is that the small-typed atomics will be implemented as must-expand intrinsics (like `int` and `long` ones are, currently). I have added a note about this to https://github.com/dotnet/runtimelab/issues/2472.